### PR TITLE
Add MaskRCNN model as a test for the wrapper approach with TF model garden code.

### PR DESCRIPTION
**Pull request by qlzh727**
# Note that this PR is more for a demo and not intend to be submitted at the moment. 

# What does this PR do?
Add MaskRCNN model as a test for the wrapper approach with TF model garden code.

1. The current class only contains the model building logic.
2. Most of the tunable params are currently backed in the function, and not exposed to use yet.
3. Verify the model building logic in the test. (will need tfm pip deps to work)

Next step:
1. Configure the input parsing logic for coco, and also anchor generation.
2. Mimic the training loop logic as model.train_step/eval_step
3. Train model e2e.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

### CODE REVIEW

File: keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+except ImportError:
+    # Didn't have tensorflow model garden package installed, and the model can't
+    # be instantiated.
+    tfm = None
+
+
+def _assert_tensorflow_model_installed():
+    if tfm is None:
+        raise ImportError(
+            'Tensor model package is not installed in the '
```
Comment (LukeWood) [inline keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py:38]: Should we provide a model name here in the error message.

"To use {MODEL}... you must install tf-models-official"

File: keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+            'environment. Please use `pip install tf-models-official` '
+            'or `pip install tf-models-nightly` to install the required '
+            'dependency package')
+
+
+def mask_rcnn(
```
Comment (LukeWood) [inline keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py:44]: do we plan to support bounding_box_format and the other KerasCV APIs (include_rescaling, etc)?

**Review by LukeWood — COMMENTED** [review]
some preliminary thoughts (i.e. do we plan to support bounding_box_format(), etc)

File: keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py (hunk 1 above)
Comment (qlzh727) [reply keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py:38]: Ack, will do.

File: keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py (hunk 2 above)
Comment (qlzh727) [reply keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py:44]: Yea, we should add those common params to make the interface more look like the existing keras cv APIs.

Comment (LukeWood) [inline keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py:44]: is `mask_rcnn` our intended API surface?

I think that by wrapping in a sub-model similar to the RetinaNet we will be able to better manage the bounding_box_format, include_rescaling, and metric evaluation during fit().

I think you'll also need to do this to get label encoding and prediction decoding working w/ model.fit(), model.evaluate(), and model.predict().

**Comment by qlzh727** [comment]
Updated the PR description. This PR is more for a demo, and not intended to be submitted at the moment.

**Comment by LukeWood** [comment]
> Updated the PR description. This PR is more for a demo, and not intended to be submitted at the moment.

Thanks for the heads up!  cool, sounds good!

File: keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py (hunk 2 above)
Comment (qlzh727) [reply keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py:44]: yea, I think we probably should take the delegate approach, that make MaskRCNN to include a tfm MaskRCNN model, and delegate all the public methods to it.

My original thought was to directly subclass the tfm MaskRCNN model, but then this requires a hard dependency to tfm, otherwise the code will fail at the package load time when tfm is not installed.

**Comment by qlzh727** [comment]
I have moved my PR to https://github.com/keras-team/keras-cv/pull/754 (since the current one was created on Master). Let's review that one. Thanks.
